src/validators/tuple.rs

```diff
@@ -296,3 +297,385 @@ impl Validator for TuplePositionalValidator {
+                }
+                Err(ValError::Omit) => (),
+                Err(err) => return Err(err),
+            },
+            None => {
+                if let Some(value) = validator.default_value(py, Some(index), extra, definitions, recursion_guard)? {
+                    output.push(value);
+                } else {
+                    errors.push(ValLineError::new_with_loc(ErrorTypeDefaults::Missing, input, index));
+                }
+            }
+        }
+        index += 1;
+        if let Some(max_length) = max_length {
+            if output.len() > max_length {
+                return Err(ValError::new(
+                    ErrorType::TooLong {
+                        field_type: "Tuple".to_string(),
+                        max_length,
+                        actual_length: output.len(),
+                        context: None,
+                    },
+                    input,
+                ));
+            }
+        }
+    }
+
+    // Generate an error if there are any extra items:
+    if let Some(result) = collection_iter.next() {
+        result?;
+        return Err(ValError::new(
+            ErrorType::TooLong {
+                field_type: "Tuple".to_string(),
+                max_length: expected_length,
+                actual_length: collection_len.unwrap_or(index),
+                context: None,
+            },
+            input,
+        ));
+    }
+
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+fn validate_tuple_variable<
+    's,
+    'data,
+    T: Iterator<Item = PyResult<&'data I>>,
+    I: Input<'data> + 'data,
+    InputT: Input<'data> + 'data,
+>(
+    py: Python<'data>,
+    input: &'data InputT,
+    extra: &Extra,
+    definitions: &'data Definitions<CombinedValidator>,
+    recursion_guard: &'s mut RecursionGuard,
+    output: &mut Vec<PyObject>,
+    errors: &mut Vec<ValLineError<'data>>,
+    validators: &[CombinedValidator],
+    variable_validator_index: Option<usize>,
+    collection_iter: &mut T,
+    collection_len: Option<usize>,
+    expected_length: usize,
+    max_length: Option<usize>,
+) -> ValResult<'data, ()> {
+    let mut index = 0;
+    let head_validators = if let Some(variable_validator_index) = variable_validator_index {
+        &validators[..variable_validator_index]
+    } else {
+        validators
+    };
+
+    // Validate the "head" items
+    validate_tuple_items(
+        py,
+        input,
+        extra,
+        definitions,
+        recursion_guard,
+        output,
+        errors,
+        head_validators.iter(),
+        &mut collection_iter.by_ref().take(head_validators.len()),
+        collection_len,
+        expected_length,
+        index,
+        max_length,
+    )?;
+
+    if let Some(variable_validator_index) = variable_validator_index {
+        let n_tail_validators = validators.len() - variable_validator_index - 1;
+        let variable_validator = &validators[variable_validator_index];
```

Comment:
If you accept my suggestion of moving head validators dance inside of this branch as per comment on line 573, this may come in useful as a way to break the validators into three chunks:

```rust
    let (head_validators, [variable_validator, tail_validators @ ..]) = validators.split_at(variable_validator_index) else {
        unreachable!("validators will always contain variable validator")
    };
```